contacts/views.py

```diff
@@ -0,0 +1,164 @@
+from django.contrib.auth.decorators import login_required
+from django.shortcuts import render
+from django.http import HttpResponse, HttpResponseNotFound
+from django.http import HttpResponseRedirect
+from contacts.models import Contact, Email, Number
+from django.contrib.auth.models import User
+
+
+def index(request):
+    return render(request, 'contacts/index.html')
+
+
+def insert(request):
+    return render(request, 'contacts/test.html')
+
+
+def add_contact(request):
+    if request.method == "POST":
+        email = request.POST['user_email']
+        first_name = request.POST['first_name']
+        last_name = request.POST['last_name']
+        note = request.POST['note']
+        contact_number = request.POST['contact_number']
+        dob = request.POST['dob']
+
+        user_id = request.session['user_id']
+        user = User.objects.get(id=user_id)
+
+        contact = Contact(first_name=first_name, last_name=last_name, dob=dob, note=note, user_id=user)
+        contact.save()
+
+        email = Email(email=email, contact_id=contact)
+        email.save()
+
+        number = Number(number=contact_number, contact_id=contact)
+        number.save()
+
+        return HttpResponseRedirect('/user_index/', {'message': 'Deleted Successfully!!'})
+    else:
+        return HttpResponse("You're looking at GET")
+
+
+def login(request):
```

Comment:
Try to use the authentication system provided by django. Don't create your own routes and views for this